src/settings/FixedDueDateSchedule/components/EditSections/components/SchedulesList/SchedulesList.test.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  within,
+  fireEvent,
+} from '@testing-library/react';
+import {
+  isArray,
+  uniqueId,
+} from 'lodash';
+
+import '../../../../../../../test/jest/__mock__';
+
+import SchedulesList from './SchedulesList';
+import ScheduleCard from '../ScheduleCard';
+
+jest.mock('../ScheduleCard', () => jest.fn(({
+  onRemoveSchedule,
+  scheduleIndex,
+}) => (
+  // eslint-disable-next-line
+  <div
+    data-testid="scheduleCard"
+    onClick={() => onRemoveSchedule(scheduleIndex)}
+  />
+)));
+jest.mock('lodash', () => {
+  const originalModule = jest.requireActual('lodash');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    uniqueId: jest.fn((prefix) => `${prefix}_uniqueId`),
+  };
+});
+
+describe('SchedulesList', () => {
+  const testTimeZone = 'UTC';
+  const testFieldsNames = [
+    'testSchedule1',
+    'testSchedule2',
+  ];
+  const testFieldsValue = [
+    {
+      key: 'testScheduleKey1',
+    },
+    {
+      key: 'testScheduleKey2',
+    },
+  ];
+  const testFields = {
+    push: jest.fn(),
+    remove: jest.fn(),
+    map: (callback) => testFieldsNames.map((name, index) => callback(name, index)),
+    value: testFieldsValue,
+  };
+
+  const labelIds = {
+    new: 'ui-circulation.settings.fDDSform.new',
+  };
+
+  afterEach(() => {
+    ScheduleCard.mockClear();
+    testFields.push.mockClear();
+    testFields.remove.mockClear();
+    uniqueId.mockClear();
+  });
+
+  const renderComponent = (testMeta = {}) => {
+    render(
+      <SchedulesList
+        timezone={testTimeZone}
+        fields={testFields}
+        meta={testMeta}
+      />
+    );
+  };
+
+  describe(('add schedule button'), () => {
+    beforeEach(() => {
+      renderComponent();
+    });
+
+    it('should render "add schedule" button', () => {
+      expect(within(screen.getByTestId('addScheduleButton')).getByText(labelIds.new))
+        .toBeVisible();
+    });
+
+    it('should add schedule', () => {
+      const addScheduleButton = screen.getByTestId('addScheduleButton');
```

Comment:
we can get it by role, so we don't need to add test id for it.

Reply:
we can but in my opinion it is better to be more specific. for example in case when we use `data-testid` and some additional button is added we will have no problems in our tests. But if we use `getByRole` our test fail because there will be two buttons on the page.

Reply:
ok, accepted.